packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_type3_staticprofile.cpp

```diff
@@ -0,0 +1,322 @@
+  if(sqrtProcs*sqrtProcs != numProcs) 
+  {
+    if(0 == mpiSession.getRank())
+      std::cerr << "Error: Invalid number of processors provided, num processors must be a perfect square." << std::endl;
+    return -1;
+  }
+  int procx = sqrtProcs;
+  int procy = sqrtProcs;
+
+  // Generate a simple 3x3 mesh
+  int nex = 3;
+  int ney = 3;
+  MeshDatabase mesh(comm,nex,ney,procx,procy);
+  mesh.print(std::cout);
+
+  int maxEntriesPerRow = 16;
+
+  // Build Tpetra Maps
+  // -----------------
+  // -- https://trilinos.org/docs/dev/packages/tpetra/doc/html/classTpetra_1_1Map.html#a24490b938e94f8d4f31b6c0e4fc0ff77
+  RCP<const MapType> row_map = rcp(new MapType(GO_INVALID, mesh.getOwnedNodeGlobalIDs(), 0, comm));
+
+  #if PRINT_VERBOSE
+  row_map->describe(*out);
+  #endif
+
+  // Type-3: Graph Construction
+  // --------------------------
+  // - Loop over every element in the mesh.
+  //   - Get list of nodes associated with each element.
+  //   - Insert node contributions if the node (row) is owned locally.
+  //   
+  auto domain_map = row_map;
+  auto range_map  = row_map;
+
+  auto owned_element_to_node_ids = mesh.getOwnedElementToNode();
+  auto ghost_element_to_node_ids = mesh.getGhostElementToNode();
+
+  RCP<TimeMonitor> timerGlobal = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("X) Global")));
+  RCP<TimeMonitor> timerElementLoopGraph = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("1) ElementLoop  (Graph)")));
+
+  RCP<GraphType> crs_graph = rcp(new GraphType(row_map, maxEntriesPerRow, Tpetra::StaticProfile));
+  
+  // Using 4 because we're using quads for this example, so there will be 4 nodes associated with each element.
+  Teuchos::Array<GlobalOrdinal> global_ids_in_row(4);
+
+  // Insert node contributions for every OWNED element:
+  for(size_t element_gidx=0; element_gidx<mesh.getNumOwnedElements(); element_gidx++)
+  {
+    // Populate global_ids_in_row:
+    // - Copy the global node ids for current owned element into an array.  
+    // - Since each element's contribution is a clique, we can re-use this for 
+    //   each row associated with this element's contribution.
+    for(size_t element_node_idx=0; element_node_idx<owned_element_to_node_ids.extent(1); element_node_idx++)
+    {
+      global_ids_in_row[element_node_idx] = owned_element_to_node_ids(element_gidx, element_node_idx);
+    }
+
+    // Add the contributions from the current row into the graph if the node is owned.
+    // - For example, if Element 0 contains nodes [0,1,4,5] and nodes 0 and 4 are owned
+    //   by the current processor, then:
+    //   - node 0 inserts [0, 1, 4, 5]
+    //   - node 1 <skip>
+    //   - node 4 inserts [0, 1, 4, 5]
+    //   - node 5 <skip>
+    for(size_t element_node_idx=0; element_node_idx<owned_element_to_node_ids.extent(1); element_node_idx++)
+    {
+      if(mesh.nodeIsOwned(global_ids_in_row[element_node_idx]))
+      {
+       crs_graph->insertGlobalIndices(global_ids_in_row[element_node_idx], global_ids_in_row());
```

Comment:
Make `global_ids_in_row` an `std::vector`, and use the following `CrsGraph::insertGlobalIndices` overload:
```
    void
    insertGlobalIndices (const GlobalOrdinal globalRow,
                         const LocalOrdinal numEnt,
                         const GlobalOrdinal inds[]);
```